### ISSUE TITLE
kiss musl/clang no libc?

### ISSUE THREAD

**Issue by scsijon** [kisslinux/kiss#67]
Sorry, couldn't find the answer in your docs, so asking here.

Can KISS work in a pure musl/clang environment with no libc, etc components?

If so i'll try it out, else like many other package systems, I shall have to forget it.

I'm looking for a new packaging system (source > compiled) I can work with for a new Puppy Linux Base.

thanks

**Comment by dylanaraps** [comment]
Do you mean running KISS without `gcc`/`binutils` and only with `clang`/`llvm`?

If so, this isn't yet possible. I'm still working on a monolithic `llvm` build which doesn't depend on `libgcc`. This involves building and bundling together `llvm`, `clang`,  `compiler-rt`, `libunwind`, `libcxx`, `libcxx-abi` and quite possibly `lld`.

**Comment by scsijon** [comment]
Yes, that is what i'm looking forward towards, you might have a look at T2's latest build iso which is getting there. I've worked with T2 before, but i'd like to work even simpler than they do.

**Comment by scsijon** [comment]
If th outcome creates compiled packages, i'm sweet as we have our own system builder, I just need compiled packages to work with, and I thought it was time we had that package building stage internally.

**Comment by scsijon** [comment]
OK, patches are not yours, ie not from your end, but internally in the source package.

And the build source package musl-1.1.23.tar.gz to give the matching output directory  something like ~/musl-1.1.23

**Comment by scsijon** [comment]
oops, go up 3 boxes, for some reason it put the answer up there

**Comment by scsijon** [comment]
> > And it would be nice if the build stage could output into a packagename-version directory as then we could keep earlier versions around for comparisons when problems with patches arise as has happened in the past when coders change internal --switches (usually names) in their source.
> 
> I'm not sure what you mean here. For reference, our `llvm`/`clang` builds use zero patches.

OK, patches are not yours, ie not from your end, but internally in the source package.

And the build source package musl-1.1.23.tar.gz to give the matching output directory something like ~/musl-1.1.23

**Comment by scsijon** [comment]
I'll monitor this thread, can you let me know when your ready please, as KISS, does seem to do just about everything else that I want.

thanks

**Comment by scsijon** [comment]
> > And it would be nice if the build stage could output into a packagename-version directory as then we could keep earlier versions around for comparisons when problems with patches arise as has happened in the past when coders change internal --switches (usually names) in their source.
> 
> I'm not sure what you mean here. For reference, our `llvm`/`clang` builds use zero patches.

sorry about that github played up and was mixing messages up.

OK, patches are not yours, ie not from your end, but internally in the source package.

And the build source package musl-1.1.23.tar.gz to give the matching output directory something like ~/musl-1.1.23

**Comment by scsijon** [comment]
back inn 15min, somethings stuffing up, i'll reboot and come back in.

**Comment by dylanaraps** [comment]
I'm testing builds now:

```
[ 17%] Building C object projects/compiler-rt/lib/builtins/CMakeFiles/clang_rt.builtins-x86_64.dir/multi3.c.o
```

**Comment by dylanaraps** [comment]
> And it would be nice if the build stage could output into a packagename-version directory as then we could keep earlier versions around for comparisons when problems with patches arise as has happened in the past when coders change internal --switches (usually names) in their source.

I'm not sure what you mean here. For reference, our `llvm`/`clang` builds use zero patches.

**Comment by scsijon** [comment]
lets see if this works now>

>        And it would be nice if the build stage could output into a packagename-version directory as then we could keep earlier versions around for comparisons when problems with patches arise as has happened in the past when coders change internal --switches (usually names) in their source.

>    I'm not sure what you mean here. For reference, our llvm/clang builds use zero patches.

sorry about that github played up and was mixing messages up.

OK, patches are not yours, ie not from your end, but internally in the source package.

And the build source package musl-1.1.23.tar.gz to give the matching output directory something like ~/musl-1.1.23

**Comment by scsijon** [comment]
I'll go away and give you some play time, i'll be on about this time tomorrow and see how you went.
Thanks for your work so far.

**Comment by scsijon** [comment]
Playing around this morning with kiss ( had an hour or so free due to unwanted rain as it's harvest time here) makes me think it's worthwhile continuing, at least for now, with trying kiss out further. I won't be on tonight, i'm rearanging my office/studio for another workstation to fit so I can dedicate one to playing with kiss. On tomorrow night instead. I've left a couple of package sugestions on your kisslinux/repo, hope you don't mind. If I proceed, there will be quite a few to add to kisslinux/ccommunity that we use with puppylinux/quirky/easyos too that may be of interest to others.

**Comment by dylanaraps** [comment]
Well, it turns out that a monolithic `llvm`/`clang` build without the dependency on `libgcc` requires a two step build. You can't use `gcc` to compile a standalone `llvm` suite so the build process would need to build a regular `llvm`/`clang` and then the jumbo one.

**Comment by scsijon** [comment]
On 06/11/19 18:50, dylan wrote:
> Well, it turns out that a monolithic `llvm`/`clang` build without the dependency on `libgcc` requires a two step build. You can't use `gcc` to compile a standalone `llvm` suite so the build process would need to build a regular `llvm`/`clang` and then the jumbo one.
> 

UM, not sure. I've been reading these >

http://releases.llvm.org/9.0.0/docs/GettingStarted.html

But this says that the first step is to get a recent GCC toolchain 
installed, even though elsewhere it talks of using clang, confusing.

http://releases.llvm.org/9.0.0/docs/CMake.html

This second one just seem to use cmake????

Doc's Index is > http://releases.llvm.org/9.0.0/docs/

and there are a lot of docs there to read!

However there are already built toolchains at the parent directory >

http://releases.llvm.org/9.0.0/

so maybe your suppose to use one of them as the toolchain to build yours???

Theres also the associated packages >

lld
lldb
llvm-lua
compiler-rt
libcxx
libcxxabi
libunwind
openmp
polly
libclc
klee
vmkit

and the

test-suite

package to deal with

Also, any updates, seem to require all these plus llvm and clang to be 
updated to match, that's a few packages to match.

I did notice some distributils are using a musl/gcc setup, and maybe 
that's why.

Do you want me to look into this, and try to make some sense of it, or 
are you following on?
